### PR TITLE
Upgrade to Franklin 2.0

### LINKED ISSUE #7: Complete install of Franklin
The basic cluster setup delivered by #6 included an initial installation of Franklin which works, but that installation is incomplete.  Here is the list of tasks:
- [ ] Set Route 53 alias (franklin.staging.azavea.com?)
- [ ] Upgrade Franklin version when https://github.com/azavea/franklin/pull/1210 has been completed
- [ ] Develop migration pathway for existing databases

The reason to delay on this and not to fold it into #6 is because the required version of Franklin is not yet complete.  @moradology has suggested that with the inclusion of pgstac into Franklin, migrations of existing STACs into the database from other Franklin deployments should become easier.  Since the idea here is that this cluster will host a single Franklin instance for the whole organization, we'll need to do a number of migrations, and if we can make them easier, then it's a benefit for anyone needing to do this.

### PR DESCRIPTION
**Pull request by jpolchlo**
[Franklin](https://github.com/azavea/franklin) has advanced to version 2.0 (at least in function if not in name), which offloads complexity onto pgstac.  This modern infrastructure will allow us to stay up to date with changes to STAC more easily, and hopefully remain open to more developments in the STAC ecosystem in the future.  This PR makes improvements to the Franklin deployment to allow for this version upgrade, including adding an init container that ensures pgstac functionality is available on the RDS database.

Closes #7

### CODE REVIEW

File: deployment/aws-terraform/application/franklin/docker/install_pgstac.py [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+def create_dashboard_schema(cursor, username: str) -> None:
+    """Create custom schema for dashboard-specific functions."""
+    cursor.execute(
+        sql.SQL(
+            "CREATE SCHEMA IF NOT EXISTS dashboard;"
+            "GRANT ALL ON SCHEMA dashboard TO {username};"
+            "ALTER ROLE {username} SET SEARCH_PATH TO dashboard, pgstac, public;"
+        ).format(username=sql.Identifier(username))
+    )
+
```
Comment (moradology) [inline deployment/aws-terraform/application/franklin/docker/install_pgstac.py:150]: torch this

File: deployment/aws-terraform/application/franklin/docker/install_pgstac.py [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+
+def create_collection_summaries_functions(cursor) -> None:
```
Comment (moradology) [inline deployment/aws-terraform/application/franklin/docker/install_pgstac.py:152]: this function can go

File: deployment/aws-terraform/application/franklin/docker/install_pgstac.py [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+        # As admin, create custom dashboard schema and functions and grant privileges to bootstrapped user
+        with psycopg.connect(stac_db_conninfo, autocommit=True) as conn:
+            with conn.cursor() as cur:
+                print("Creating dashboard schema...")
+                create_dashboard_schema(cursor=cur, username=username)
+
+                print(
+                    "Creating functions for summarizing default collection datetimes and cog_default statistics..."
+                )
+                create_collection_summaries_functions(cursor=cur)
```
Comment (moradology) [inline deployment/aws-terraform/application/franklin/docker/install_pgstac.py:377]: This can go too

**Comment by jpolchlo** [comment]
I have this working at https://franklin.compute.azavea.com

**Pull request merged**